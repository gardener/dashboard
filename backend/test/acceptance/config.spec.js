//
// Copyright (c) 2020 by SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

'use strict'

module.exports = function ({ agent }) {
  /* eslint no-unused-expressions: 0 */

  it('should return the frontend configuration', async function () {
    const res = await agent
      .get('/config.json')

    expect(res).to.have.status(200)
    expect(res).to.be.json
    expect(res.body).to.have.property('helpMenuItems').that.is.an('array')
    expect(res.body.helpMenuItems).to.have.length(3)
    expect(res.body.landingPageUrl).to.equal('https://gardener.cloud/')
  })
}
