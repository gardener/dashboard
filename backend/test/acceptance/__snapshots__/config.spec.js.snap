// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`config should return the frontend configuration 1`] = `
Array [
  Array [
    Object {
      ":authority": "kubernetes:6443",
      ":method": "get",
      ":path": "/api/v1/namespaces/kube-system/configmaps/cluster-identity",
      ":scheme": "https",
      "authorization": "Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50OmdhcmRlbjpkZWZhdWx0In0.-4rSuvvj5BStN6DwnmLAaRVbgpl5iCn2hG0pcqx0NPw",
    },
  ],
]
`;

exports[`config should return the frontend configuration 2`] = `
Object {
  "apiServerUrl": "https://kubernetes.external.foo.bar",
  "clusterIdentity": "test-id",
  "features": Object {
    "terminalEnabled": true,
  },
  "helpMenuItems": Array [
    Object {
      "icon": "description",
      "title": "Getting Started",
      "url": "https://gardener.cloud/about/",
    },
    Object {
      "icon": "mdi-slack",
      "title": "Feedback",
      "url": "https://kubernetes.slack.com/messages/gardener/",
    },
    Object {
      "icon": "mdi-bug",
      "title": "Issues",
      "url": "https://github.com/gardener/dashboard/issues/",
    },
  ],
  "landingPageUrl": "https://gardener.cloud/",
  "primaryLoginType": "oidc",
  "serviceAccountDefaultTokenExpiration": 42,
}
`;
